src/QsCompiler/Tests.Compiler/TestCases/LinkingTests/ClassicalControl.qs

```diff
@@ -909,4 +909,95 @@ namespace Microsoft.Quantum.Testing.ClassicalControl {
             Bar(r1);
         }
     }
+}
+
+// =================================
+
+// Inequality with ApplyConditionally
+namespace Microsoft.Quantum.Testing.ClassicalControl {
+    open SubOps;
+
+    operation Bar(r : Result) : Unit { }
+
+    operation Foo() : Unit {
+        let r1 = Zero;
+        let r2 = Zero;
+
+        if (r1 != r2) {

```

Comment:
Am I correct in that this basically gets translated into "if (r1 == r2) { SubOp1(); } else { Bar(r1); }", and the replaced accordingly?

Reply:
Yes, that's right. If there was no 'else' clause to flip with the conditional body, the missing 'else' is treated like a NoOp.